src/natcap/invest/habitat_quality.py

```diff
@@ -188,281 +191,326 @@
-                '.tif', '_aligned.tif'))
+            intermediate_output_dir, os.path.basename(lulc_path).replace(
+                os.path.splitext(lulc_path)[1],
+                f'_aligned{file_suffix}.tif'))
         for threat in threat_dict:
             threat_path = threat_path_dict['threat' + lulc_key][threat]
             if threat_path in lulc_and_threat_raster_list:
                 aligned_threat_path = os.path.join(
-                    inter_dir, os.path.basename(threat_path).replace(
-                        '.tif', '_aligned.tif'))
+                    intermediate_output_dir,
+                    os.path.basename(threat_path).replace(
+                        os.path.splitext(threat_path)[1],
+                        f'_aligned{file_suffix}.tif'))
+                aligned_bound_threat_path = os.path.join(
+                    intermediate_output_dir,
+                    os.path.basename(aligned_threat_path).replace(
+                        f'aligned{file_suffix}',
+                        f'aligned_bound{file_suffix}'))
+                # Use these updated threat raster paths in future calculations
                 threat_path_dict['threat' + lulc_key][threat] = (
-                    aligned_threat_path)
-
-                # Iterate though the threat raster and update pixel values
-                # as needed so that:
-                #  * Nodata values are replaced with 0
-                #  * Anything other than 0 or nodata is replaced with 1
-                LOGGER.info('Preprocessing threat values for %s',
-                            aligned_threat_path)
-                threat_nodata = pygeoprocessing.get_raster_info(
-                    aligned_threat_path)['nodata'][0]
-                threat_raster = gdal.OpenEx(aligned_threat_path,
-                                            gdal.OF_RASTER | gdal.GA_Update)
-                threat_band = threat_raster.GetRasterBand(1)
-                for block_offset in pygeoprocessing.iterblocks(
-                        (aligned_threat_path, 1), offset_only=True):
-                    block = threat_band.ReadAsArray(**block_offset)
-
-                    # First check if we actually need to set anything.
-                    # No need to perform unnecessary writes!
-                    if set(numpy.unique(block)) == set([0, 1]):
-                        continue
-
-                    zero_threat = numpy.isclose(block, threat_nodata)
-                    block[zero_threat] = 0
-                    block[~numpy.isclose(block, 0)] = 1
-
-                    threat_band.WriteArray(
-                        block, yoff=block_offset['yoff'],
-                        xoff=block_offset['xoff'])
-                threat_band = None
-                threat_raster = None
+                    aligned_bound_threat_path)
+
+                # Bound threat raster values to 0 <= x <= 1
+                bound_threat_values_task = task_graph.add_task(
+                    _bound_raster_values,
```

Comment:
Can you specifically name the `func=` argument here? It helps when you're reading that you can see `_bound_raster_values` is a function. Also with a newer version of Python I'd love to upgrade `add_task` to forbid positional parameters as a kind of syntactic salt. If you've already got `func=` it won't be a problem for an upgrade.